### PR TITLE
Feature/renewing gateway module interface

### PR DESCRIPTION
**Pull request by martinichka**
Integrated new gateway module interface for robust communication with the LoRa gateway module (LG8271/LG9271).

### CODE REVIEW

File: .ci/install-azure-cli.sh [hunk 1]
```diff
@@ -5,4 +5,4 @@ set -e -o pipefail
 apt-key adv --keyserver packages.microsoft.com --recv-keys 52E16F86FEE04B979B07E28DB02C46DF417A0893
```
Comment (htdvisser) [inline .ci/install-azure-cli.sh]: Replace this with `curl -L https://packages.microsoft.com/keys/microsoft.asc | apt-key add -`

File: .ci/install-azure-cli.sh [hunk 2]
```diff
@@ -5,4 +5,4 @@ set -e -o pipefail
...
 echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" > /etc/apt/sources.list.d/azure-cli.list
 apt-get update
-apt-get install -y azure-cli jq
+apt-get install -y azure-cli jq --allow-unauthenticated
```
Comment (htdvisser) [inline .ci/install-azure-cli.sh]: Undo this

File: firmware/src/app_wifi.c [hunk 3]
```diff
@@ -446,7 +446,10 @@ void APP_WIFI_Tasks(void)
             macAddress = TCPIP_STACK_NetAddressMac(netHandle);
             if(macAddress)
             {
-                _changeState(STATE_PRESCAN);
+                // In the past here was switched to STATE_PRESCAN. But that cause another thread (lora_task)
+                // to fail with a hard-fault. Go here ot STATE_WAIT_CONFIG instead such that for sure either
+                // the EVENT_SWITCH_TO_AP_MODE or EVENT_SWITCH_TO_INFO_MODE events has been proccessed.
+                _changeState(STATE_WAIT_CONFIG);
```
Comment (htdvisser) [inline firmware/src/app_wifi.c]: This change seems to be unrelated to the title of this Pull Request

Comment (martinichka) [reply firmware/src/app_wifi.c]: This is one of the three essential fixes to make it more robust. This is a prerequisite for the integration of the gateway module interface library (without this fix it will run into a hard-fault).

So it either comes with this PR or before it in a seperate PR.

Comment (KrishnaIyer) [reply firmware/src/app_wifi.c]: Yeah i know this one from Bitbucket. I'm ok with including it here.

**Comment by htdvisser** [comment]
Submodule `firmware/vendor/gateway-module-interface` is not on the latest commit, maybe fix that before merging.

**Comment by martinichka** [comment]
Submodule firmware/vendor/gateway-module-interface updated to the latest commit

**Comment by KrishnaIyer** [comment]
Tests seem to work well. No impact on the WiFi module. All 3 gateways (2 faulty, 1 normal with `1.0.4`) seem to have survived the reboot and sending uplinks normally. I'm gonna merge this so that people can test it.
